main.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: false
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    x = 0
+    while x < self.length
+      yield(self[x])
+      x += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    x = 0
+    while x < self.length
+      yield(self[x], x)
+      x += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(method) if !block_given?
+
+    arr = []
+    x = 0
+    while x < self.length
+      if yield(self[x])
+        arr << self[x]
+      end
+      x += 1
+    end
```

Comment:
This method should return a enumerable, when you didn't pass a block to it.

But in your case, If you didn't pass a block it throws error `wrong number of arguments (given 0, expected 1)`